stdlib/public/core/ArrayShared.swift

```diff
@@ -222,17 +198,20 @@ extension _ArrayBufferProtocol {
   /// copying when it isUniquelyReferenced.
   @inline(never)
   @inlinable // @specializable
-  internal mutating func _arrayOutOfPlaceUpdate<Initializer>(
+  internal mutating func _arrayOutOfPlaceUpdate(
     _ dest: inout _ContiguousArrayBuffer<Element>,
     _ headCount: Int, // Count of initial source elements to copy/move
     _ newCount: Int,  // Number of new elements to insert
-    _ initializeNewElements: Initializer
-  ) where
-    Initializer: _PointerFunction,
-    Initializer.Element == Element {
+    _ initializeNewElements: 
+        ((UnsafeMutablePointer<Element>, _ count: Int) -> ())? = nil
+  ) {
 
     _sanityCheck(headCount >= 0)
     _sanityCheck(newCount >= 0)
+    
+    let initializeNewElements = initializeNewElements ?? { ptr, count in
```

Comment:
What if the closure parameter was non-optional, with this as the default value?

Reply:
It makes the function signature pretty gnarly but maybe it's worth it?